docker/developers/install-local-nvm.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -e
+
+# This script works alongside the assumptions make in the Dockerfile to support
+# updating a containerized install of NVM/Node/npm for use within the container.
+
+if [ "$(hostname)" != 'install-nvm' ]; then
+	echo "This is meant to run inside the docker compose context."
+	echo "Run with: 'docker compose run install-nvm'"
+	exit 1
+fi
+
+echo "Installing nvm into the Docker 'nvm' volume."
+
+# Copy over a pre-built copy of nvm
+if [ ! -e ${HOME}/.nvm/package.json ]; then
+  cp -r /opt/base-nvm/* ${HOME}/.nvm/.
+fi
+
+# Initialize NVM
+. ${HOME}/.nvm/nvm.sh
+
+# Determine the desired node version
+NODE_VERSION=`jq -r .engines.node < apps/package.json | sed "s/[\^]//g"`
+echo " - Installing node v${NODE_VERSION}"
```

Comment:
The semver in `package.json` is for the highest minor of 18, instead of reading the version from `package.json` we should probably look at [.nvmrc](https://github.com/code-dot-org/code-dot-org/blob/staging/.nvmrc). If we change directories to `apps`, we can run `nvm install`, or we can read `.nvmrc` to get the version. I also think we can unpin the version in `.nvmrc` but that is a separate issue.